packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts

```diff
@@ -59,6 +68,10 @@ export class StepFunctionsStartExecution extends sfn.TaskStateBase {
       throw new Error('Task Token is required in `input` for callback. Use JsonPath.taskToken to set the token.');
     }
 
+    if (this.props.associateWithParent && props.input && props.input.type !== sfn.InputType.OBJECT) {
+      throw new Error('Could not associate child execution with parent execution because input is taken directly from a JSON path (or a value for the `input` property was not provided). Use `sfn.TaskInput.fromObject` instead.');
```

Comment:
```suggestion
      throw new Error('Could not enable `associateWithParent` because `input` is taken directly from a JSON path. Use `sfn.TaskInput.fromObject` instead.');
```